pyflakes/test/test_other.py

```diff
@@ -1766,6 +1767,88 @@ def test_assign_expr(self):
         ''')
 
 
+class TestUnusedFunction(TestCase):
+    """
+    Tests for warning about unused functions.
+    """
+
+    def test_unusedFunction(self):
+        """
+        Warn when a function inside a function is defined but never used.
+        """
+        self.flakes('''
+        def a():
+            def b():
+                pass
+        ''', m.UnusedFunction)
+
+    def test_unusedUnderscoreFunction(self):
+        """
+        Don't warn when the magic "_" (underscore) name is unused.
+        See issue #202.
+        """
+        self.flakes('''
+        def a():
+            def _():
+                pass
+        ''')
+
+    def test_usedDecoratedFunction(self):
+        """
+        Don't warn when the function is decorated because decorators can do
+        anything, like copy it into global state.
+        """
+        self.flakes('''
+        from somewhere import decorator
+
+        def a():
+            @decorator
+            def b():
+                pass
+        ''')
+
+
+class TestUnusedClass(TestCase):
+    """
+    Tests for warning about unused classes.
+    """
+
+    def test_unusedClass(self):
```

Comment:
This test could be moved to a new PR and `UnusedClass` introduced in a non-controversial manner.